### ISSUE TITLE
[Bug Report] Fails to load model on specific GPU

### ISSUE THREAD

**Issue by slavachalnev** [neelnanda-io/TransformerLens#249]
**Describe the bug**
Error when loading the model to a specific device (e.g. `device='cuda:0'`)

**Code example**
```python
from transformer_lens import HookedTransformer

model = HookedTransformer.from_pretrained("gpt2-small", device="cuda:0")
```
Results in error:
```
Traceback (most recent call last):
  File "a.py", line 8, in <module>
    model = HookedTransformer.from_pretrained("gpt2-small", device="cuda:0")
  File "/home/slava/safety/TransformerLens/transformer_lens/HookedTransformer.py", line 758, in from_pretrained
    model = cls(cfg, **model_kwargs)
  File "/home/slava/safety/TransformerLens/transformer_lens/HookedTransformer.py", line 161, in __init__
    HookedTransformer.move_model_modules_to_device(self)
  File "/home/slava/safety/TransformerLens/transformer_lens/HookedTransformer.py", line 666, in move_model_modules_to_device
    model.embed.to(devices.get_device_for_block_index(0, model.cfg))
  File "/home/slava/safety/TransformerLens/transformer_lens/utilities/devices.py", line 19, in get_device_for_block_index
    return torch.device(device, index // layers_per_device)
RuntimeError: type (string) must not include an index because index was passed explicitly: cuda:0
```

**System Info**
Trafsformer Lens installed from source
Linux Ubuntu 20.04.2
Nvidia 3090 GPU
Python 3.8.10

**Additional context**
I think the way to fix this is to modify `get_device_for_block_index`. Replace this:
```python
if isinstance(device, torch.device):
        device = device.type
```
with this:
```python
device = torch.device(device).type
```

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/neelnanda-io/TransformerLens/issues) in the repo (**required**)

**Comment by slavachalnev** [comment]
I am currently working on #246.

That should fix this issue and #217. I think I'll also fix #248 wile I'm at it.

**Comment by jbloomAus** [comment]
solved by #250
